src/stores/ui.ts

```diff
@@ -1,33 +1,148 @@
   state: (): UiState => ({
-    instructionsSnackbarState: 'not_shown',
+    instructionsSnackbarState: NOT_SHOWN,
+    isFilterVisible: false,
+    isFilterDismissed: false,
+    isDesktopLayout: false,
+    isMobileUa: true,
   }),
 
   getters: {
-    areInstructionsVisible(state) {
-      return state.instructionsSnackbarState === 'visible'
+    areInstructionsVisible(state): boolean {
+      return state.instructionsSnackbarState === VISIBLE
     },
   },
 
   actions: {
     showInstructionsSnackbar() {
-      if (this.instructionsSnackbarState === 'not_shown') {
-        this.instructionsSnackbarState = 'visible'
+      if (this.instructionsSnackbarState === NOT_SHOWN) {
+        this.instructionsSnackbarState = VISIBLE
       }
     },
     hideInstructionsSnackbar() {
-      this.instructionsSnackbarState = 'dismissed'
+      this.instructionsSnackbarState = DISMISSED
+    },
+    /**
+     * Given a list of key value pairs of UI state parameters and their states,
+     * populate the store state to match the cookie.
+     *
+     * Values stored in the cookie are stored across sessions and are modified
+     * if the breakpoints change.
+     *
+     * @param cookies - mapping of UI state parameters and their states.
+     */
+    initFromCookies(cookies: UiStateCookie) {
+      this.isDesktopLayout = cookies.isDesktopLayout ?? false
+      this.isMobileUa = cookies.isMobileUa ?? false
+      this.isFilterDismissed = cookies.isFilterDismissed ?? false
+      this.isFilterVisible = this.isDesktopLayout
+        ? !this.isFilterDismissed
+        : false
+    },
+    /**
+     * Sets the breakpoint and ua state. We have to set the cookie value inside
+     * the `setup` function because `app.$cookies` is not available outside it.
+     * This is why we return the object that needs to be saved in the cookies.
+     * @param isDesktopLayout - whether the layout is desktop (`lg` with the `new_header`
+     * and `md` with the `old_header`).
+     * @param isMobileUa - whether the UA is `mobile` or not.
+     */
+    updateBreakpoint(
+      isDesktopLayout: boolean,
+      isMobileUa: boolean | undefined
+    ): ScreenParameters | undefined {
+      if (
+        this.isDesktopLayout === isDesktopLayout &&
+        typeof isMobileUa === 'undefined'
+      )
+        return
+      this.isDesktopLayout = isDesktopLayout
+      // Only update `isMobileUa` if the value is provided
+      if (typeof isMobileUa !== 'undefined') {
+        this.isMobileUa = isMobileUa
+      }
+      return {
+        isDesktopLayout: this.isDesktopLayout,
+        isMobileUa: this.isMobileUa,
+      }
+    },
+    /**
+     * Sets the filter state based on the `visible` parameter.
+     * Closing the filters sets the state to `dismissed` on desktop,
+     * but `shown_on_desktop` on mobile.
+     *
+     * The calling function must be inside the `setup` function and
+     * must set the cookies if the return value is not `null`.
```

Comment:
Is this because of the `$cookies` limitation as well?

Reply:
Yes :( I tried passing the `app.$cookies.set` as a function parameter to this action to overcome this limitation, but it didn't work, I still got the `this function must be called from inside the setup function` warning.